applications/Chat/evaluate/metric.py

```diff
@@ -0,0 +1,152 @@
+import jieba
+from nltk.translate.bleu_score import sentence_bleu
+from rouge_chinese import Rouge
+from bert_score import score
+from sklearn.metrics import precision_score, recall_score, f1_score
+
+
+def bleu_score(preds: list, targets: list) -> dict:
+    bleu_score = {"bleu1": 0, "bleu2": 0, "bleu3": 0, "bleu4": 0}
+    cumulate_bleu1 = cumulate_bleu2 = cumulate_bleu3 = cumulate_bleu4 = 0
+
+    # calculate scores
+    for pred, target in zip(preds, targets):
+        pred_list = []
+        target_list = []
+        pred_list = (' '.join(jieba.cut(pred))).split()
+        target_list.append((' '.join(jieba.cut(target))).split())
+
+        bleu1 = sentence_bleu(target_list, pred_list, weights=(1, 0, 0, 0))
+        bleu2 = sentence_bleu(target_list, pred_list, weights=(0, 1, 0, 0))
+        bleu3 = sentence_bleu(target_list, pred_list, weights=(0, 0, 1, 0))
+        bleu4 = sentence_bleu(target_list, pred_list, weights=(0, 0, 0, 1))
```

Comment:
As far as I understand, if you want to calculate 3-gram BLEU, you should consider 1-gram, 2-gram and 3-gram at the same time.

See the example here:
https://www.nltk.org/_modules/nltk/translate/bleu_score.html

Also you can also put weights in a list to avoid repetition code.